src/fdb5/rules/Schema.cc

```diff
@@ -52,30 +52,30 @@ const Rule*  Schema::ruleFor(const Key& dbKey, const Key& idxKey) const {
 }
 
 void Schema::expand(const metkit::mars::MarsRequest &request, ReadVisitor &visitor) const {
-    Key full(registry());
-    std::vector<Key> keys(3);
+    CanonicalKey full(registry());
+    std::vector<ApiKey> keys(3);
     for (auto& k : keys) k.registry(registry());
 
-    for (std::vector<Rule *>::const_iterator i = rules_.begin(); i != rules_.end(); ++i ) {
+    for (Rule* r : rules_) {
 		// eckit::Log::info() << "Rule " << **i <<  std::endl;
 		// (*i)->dump(eckit::Log::info());
-		(*i)->expand(request, visitor, 0, keys, full);
+		r->expand(request, visitor, 0, keys, full);
     }
 }
 
-void Schema::expand(const Key &field, WriteVisitor &visitor) const {
-    Key full(registry());
-    std::vector<Key> keys(3);
+void Schema::expand(const CanonicalKey& field, WriteVisitor &visitor) const {
```

Comment:
This feels the wrong way around - we are expanding a canonicalised key, and producing a non-canonicalised key?